tool_quality_checks/correct_import_file.py

```diff
@@ -0,0 +1,244 @@
+        field_names.append(layer_defn.GetFieldDefn(n).name)
+
+    for count, in_feat in enumerate(in_layer):
+        content = {}
+        for field_name in field_names:
+            content[field_name] = in_feat.GetField(field_name)
+
+        geom = in_feat.GetGeometryRef()
+        if geom == None:
+            logger.warning("FID {} has no geometry.".format(count))
+            lost_features.append(in_feat.GetFID())
+            continue
+
+        if (
+            geom.GetGeometryName() == "MULTIPOLYGON"
+            or geom.GetGeometryName() == "MULTILINESTRING"
+        ):
+            for geom_part in geom:
+                addPolygon(geom_part.ExportToWkb(), content, out_layer)
+        else:
+            addPolygon(geom.ExportToWkb(), content, out_layer)
+
+    return lost_features
+
+
+def addPolygon(simple_polygon, content, out_lyr):
+    featureDefn = out_lyr.GetLayerDefn()
+
+    polygon = ogr.CreateGeometryFromWkb(simple_polygon)
+    out_feat = ogr.Feature(featureDefn)
+    out_feat.SetGeometry(polygon)
+
+    for key, value in content.items():
+        out_feat.SetField(key, value)
+
+    out_lyr.CreateFeature(out_feat)
+
+
+def fix_geometry(geometry):
+    geom_name = geometry.GetGeometryName()
+    # check pointcount if linestring
+    if "LINESTRING" in geom_name:
+        if geometry.GetPointCount() == 1:
+            logger.warning("Geometry point count of linestring = 1")
+            return geometry, False
+        else:
+            pass
+
+    # solve self intersections
+    if "POLYGON" in geom_name:
+        try:
+            geometry = geometry.Buffer(0)
+        except Exception as e:
+            # RuntimeError: IllegalArgumentException: Points of LinearRing do not form a closed linestring
+            logger.warning(e)
+            return geometry, False
+
+    # We think this part is not used or creates an error. So just leave it out for a while
+    # # check slivers
+    # if not geometry.IsValid():
+    #     perimeter = geometry.Boundary().Length()
+    #     area = geometry.GetArea()
+
+    #     sliver = float(perimeter / area)
+
+    #     if sliver < 1:
+    #         wkt = geometry.ExportToWkt()
+    #         geometry = ogr.CreateGeometryFromWkt(wkt)
+
+    return geometry, geometry.IsValid()
+
+
+def correct(in_layer, layer_name="", epsg=28992):
+    """
+        This function standardizes a vector layer:
+            1. Multipart to singleparts
+            2. 3D polygon to 2D polygon
+            3. Reprojection
+            4. Fix for self intersections
+            
+        INPUT: layer, layer_name
+        OUTPUT: out_datasource, layer_name
+      """
+
+    # retrieving lost features
+    lost_features = []
+    in_feature_count = in_layer.GetFeatureCount()
+
+    # Get inspatial reference and geometry from in shape
+    geom_type = in_layer.GetGeomType()
+    in_spatial_ref = in_layer.GetSpatialRef()
+    in_layer.ResetReading()
+
+    if layer_name != "":  # lower layer_name
```

Comment:
```suggestion
    if not layer_name:  # lower layer_name
```

Reply:
Eigenlijk is de hele "if" niet nodig: ook als 'ie een lege string is kan je prima checken of 'ie korter is dan 50.